src/libraries/System.Collections/tests/Generic/SortedList/SortedList.Generic.Tests.cs

```diff
@@ -259,6 +259,59 @@ public void SortedList_Generic_ContainsValue_DefaultValuePresent(int count)
 
         #endregion
 
+        #region GetKeyAtIndex
+
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void SortedList_Generic_GetKeyAtIndex_EveryIndex(int count)
+        {
+            SortedList<TKey, TValue> dictionary = (SortedList<TKey, TValue>)GenericIDictionaryFactory(count);
+            Assert.All(Enumerable.Range(0, count), index =>
+            {
+                Assert.Equal(index, dictionary.IndexOfKey(dictionary.GetKeyAtIndex(index)));
+            });
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void SortedList_Generic_GetKeyAtIndex_OutOfRangeIndicies(int count)
+        {
+            SortedList<TKey, TValue> dictionary = (SortedList<TKey, TValue>)GenericIDictionaryFactory(count);
+            Assert.Throws<ArgumentOutOfRangeException>(() => dictionary.GetKeyAtIndex(-1));
+            Assert.Throws<ArgumentOutOfRangeException>(() => dictionary.GetKeyAtIndex(int.MinValue));
+            Assert.Throws<ArgumentOutOfRangeException>(() => dictionary.GetKeyAtIndex(count));
+            Assert.Throws<ArgumentOutOfRangeException>(() => dictionary.GetKeyAtIndex(count + 1));
+        }
+
+        #endregion
+
+        #region GetValueAtIndex
+
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void SortedList_Generic_GetValueAtIndex_EveryIndex(int count)
+        {
+            // Assumes no duplicate elements contained in the dictionary returned by GenericIDictionaryFactory
+            SortedList<TKey, TValue> dictionary = (SortedList<TKey, TValue>)GenericIDictionaryFactory(count);
+            Assert.All(Enumerable.Range(0, count), index =>
+            {
+                Assert.Equal(index, dictionary.IndexOfValue(dictionary.GetValueAtIndex(index)));
```

Comment:
I assume this would require unique values, but if the test is passing then I guess it's fine.